apps/core/downloadmanager.cpp

```diff
@@ -132,6 +133,9 @@ void DownloadManager::recvObserve(const QString message, const QVariant data)
         } else if (isMyApp(targetPath)) {
             finishMyAppDownload(targetPath);
         }
+        if (isPrivateMode) {
+            m_transferClient->clearTransfer(m_download2transferMap.value(downloadId));
+        }
```

Comment:
Small downloads disappear now really quickly. I doubt that people find downloaded files from the file system. Instead of immediately clearing, I think it would be nicer to gather a list of completed private downloadIds over here. Then upon clearing of PrivateTabModel (tab count reaching zero and destruction of the model) clear also private downloads from the DownloadManager. How do you see this?

Briefly chatted this with @pvuorela as well.